etc/spack/defaults/config.yaml

```diff
@@ -75,6 +75,11 @@ config:
   misc_cache: ~/.spack/cache
 
 
+  # Timeout in seconds used for downloading sources etc. This can be
+  # increased in case servers are too slow to answer. 0 means no timeout.
+  download_timeout: 10
```

Comment:
My only thought is that this should be called `connect_timeout`: to me, "download timeout" implies that Spack might quit in the middle of a download.

I was hoping to add something less-manual so that users wouldn't have to tweak this for packages (e.g. adding attributes to packages which allow setting per-package timeouts), but I think this is a great option for now.

Reply:
> My only thought is that this should be called `connect_timeout`: to me, "download timeout" implies that Spack might quit in the middle of a download.

I renamed it to `connect_timeout`.

> I was hoping to add something less-manual so that users wouldn't have to tweak this for packages (e.g. adding attributes to packages which allow setting per-package timeouts), but I think this is a great option for now.

Good point. I have also pushed two new commits reworking the `curl_options` into something a bit more generic and added support for a new `timeout` option there that allows individual packages to override the timeout. Let me know what you think.